proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/PlayerListItem.java

```diff
@@ -25,11 +25,16 @@
   public static final int UPDATE_DISPLAY_NAME = 3;
   public static final int REMOVE_PLAYER = 4;
   private int action;
-  private final List<Item> items = new ArrayList<>();
+  private List<Item> items;
 
   public PlayerListItem(int action, List<Item> items) {
     this.action = action;
-    this.items.addAll(items);
+    (this.items = new ArrayList<>(items.size())).addAll(items);
+  }
+
+  public PlayerListItem(Void unused, int action, List<Item> items) {
```

Comment:
This looks like a really odd hack to distinct the two constructors. I think it must be settled whether or not we make a defensive copy to prevent side-effects.

Reply:
Adding to this, would remove this constructor and pass an empty list to the previous constructor instead. I don't think these kinds of micro-optimizations yield observable performance gains.

Reply:
Why was it done like this instead of just using a boolean 'cloneList' for example?

Reply:
I'm not sure, but maybe a static method could be a more descriptive interface